Fabric.Identity.API/scripts/tests/CatalystDosIdentity.tests.ps1

```diff
@@ -0,0 +1,461 @@
+        }
+    }
+
+    Context 'Invalid request - invalid json' {
+        It 'New Registration should return an error when invalid client json' {
+            $authUrl = [System.Uri]'https://some.Server/identity'
+
+            $newClient = @{}
+            $newClient.Add("clientId", "cliTestClient")
+            $newClient.Add("clientName", "cli Test Client Name")
+            $newClient.Add("requireConsent", "false")
+
+            $jsonClient = $newClient | ConvertTo-Json
+
+            Mock -ModuleName CatalystDosIdentity -CommandName Invoke-RestMethod {
+                throw  New-Object -TypeName "System.Net.WebException" -ArgumentList "There was an error registering client"
+            } `
+                -ParameterFilter {
+                $uri -match "$authUrl"
+            }
+            {New-ClientRegistration -identityUrl $authUrl -body $jsonClient -accessToken "goodToken"} | Should -Throw "There was an error registering client" -ExceptionType System.Net.WebException
+        }
+    }
+    Context "Invalid request - bad token" {
+        It 'new registration should return an error when bad token' {
+            $authUrl = [System.Uri]'https://some.Server/identity'
+
+            $newClient = New-ClientCredentialsClientBody `
+                -clientId "cliTestClient" `
+                -clientName "cli Test Client Name" `
+                -allowedScopes @("fabric/identity.manageresources", "fabric/authorization.read", "fabric/authorization.write", "fabric/authorization.manageclients")
+
+            $jsonClient = $newClient | ConvertTo-Json
+
+            Mock -ModuleName CatalystDosIdentity -CommandName Assert-WebExceptionType -Verifiable -MockWith {
+                return $false
+            } -ParameterFilter {$typeCode -eq 409}
+
+            Mock -ModuleName CatalystDosIdentity -CommandName Invoke-RestMethod -Verifiable -MockWith {
+                throw  New-Object -TypeName "System.Net.WebException" -ArgumentList "(401) Unauthorized"
+            } -ParameterFilter {$Method -match 'Post' -and $uri -match "$authUrl/api/client"}
+
+            {New-ClientRegistration -identityUrl $authUrl -body $jsonClient -accessToken "badToken"} | Should -Throw "There was an error registering client" -ExceptionType System.Net.WebException
+        }
+    }
+}
+
+describe 'New-ClientCredentialsClientBody Unit Tests' -tag 'Unit' {
+    Context 'Valid Object' {
+        It 'New-ClientCredentialsClientBody should return a valid object with required parameters' {
+            $myClientId = "validClientCredentialsClient"
+            $myClientName = "Valid client credentials client name"
+
+            $response = New-ClientCredentialsClientBody -clientId $myClientId -clientName $myClientName -allowedScopes @("fabric/authorization.read", "dos/metadata", "fabric.profile")
+            $response | Should -not -Be :null
```

Comment:
doesn't affect function, but we probably should camel case Should -Not -Be, instead of Should -not -Be or Should -not -be